qiskit_experiments/calibration_management/calibration_utils.py

```diff
@@ -12,51 +12,137 @@
 
-    for schedule in schedules:
-        if _used_in_calls(schedule_name, schedule):
-            caller_names.add(schedule.name)
+    return callers
 
-    return caller_names
 
+def _referred_by(schedule_name: str, dag: rx.PyDiGraph) -> Set[str]:
+    """Return all the schedules that refer to this schedule by name."""
+    referred_by = set()
+    for predecessor in dag.predecessors(dag.nodes().index(schedule_name)):
+        referred_by.add(predecessor)
+        referred_by.update(_referred_by(predecessor, dag))
 
-def _used_in_calls(schedule_name: str, schedule: ScheduleBlock) -> bool:
-    """Recursively find if the schedule calls a schedule with name ``schedule_name``.
+    return referred_by
+
+
+def validate_channels(schedule: ScheduleBlock) -> Set[Parameter]:
+    """Validate and get the parameters in the channels of the schedule.
+
+    Channels implicitly defined in references are ignored.
 
     Args:
-        schedule_name: The name of the callee to identify.
-        schedule: The schedule to parse.
+        schedule: The schedule for which to get the parameters in the channels.
 
     Returns:
-        True if ``schedule``calls a ``ScheduleBlock`` with name ``schedule_name``.
+        The set of parameters explicitly defined in the schedule.
+
+    Raises:
+        CalibrationError: If a channel is parameterized by more than one parameter.
+        CalibrationError: If the parameterized channel index is not formatted properly.
     """
-    blocks_have_schedule = False
-
-    for block in schedule.blocks:
-        if isinstance(block, Call):
-            if block.subroutine.name == schedule_name:
-                return True
-            else:
-                blocks_have_schedule = blocks_have_schedule or _used_in_calls(
-                    schedule_name, block.subroutine
+
+    # The channel indices need to be parameterized following this regex.
+    __channel_pattern__ = r"^ch\d[\.\d]*\${0,1}[\d]*$"
+
+    param_indices = set()
+
+    # Schedules with references do not explicitly have channels. This needs special handling.
+    if schedule.is_referenced():
+        for block in schedule.blocks:
+            if isinstance(block, ScheduleBlock):
+                param_indices.update(validate_channels(block))
+
+        return param_indices
+
+    for ch in schedule.channels:
+        if isinstance(ch.index, ParameterExpression):
+            if len(ch.index.parameters) != 1:
+                raise CalibrationError(f"Channel {ch} can only have one parameter.")
+
+            param_indices.add(ch.index)
+            if re.compile(__channel_pattern__).match(ch.index.name) is None:
+                raise CalibrationError(
+                    f"Parameterized channel must correspond to {__channel_pattern__}"
                 )
 
-        if isinstance(block, ScheduleBlock):
-            blocks_have_schedule = blocks_have_schedule or _used_in_calls(schedule_name, block)
+    return param_indices
+
+
+def reference_info(
```

Comment:
You can cache this function for speed up.

Reply:
Done here: [2361e03](https://github.com/Qiskit/qiskit-experiments/pull/889/commits/2361e03219a326fd9de04706a35ef518486408fa)